crates/circuit/src/dag_circuit.rs

```diff
@@ -4787,6 +4765,12 @@ impl DAGCircuit {
         &self.qargs_interner
     }
 
+    /// Returns an mutable view of the Interner used for Qargs
+    #[inline(always)]
+    pub fn qargs_interner_mut(&mut self) -> &mut Interner<[Qubit]> {
+        &mut self.qargs_interner
+    }
+
```

Comment:
This is newly added in this PR, and I feel like we'd decided that we don't want to expose this, because if you have the mutable reference, you can `::mem::swap` in an invalid interner and easily break the data coherence of the whole `DAGCircuit` a bit too easily.

At any rate, if there's a new compelling reason for it (and one that `merge_qargs` doesn't already cover - I'd assume that that's sufficient and faster for `disjoint_utils`), can we do it in the relevant PR?

Reply:
https://github.com/Qiskit/qiskit/pull/14313/files/f1bc20b4f052553680dd79c2cdb3f26da29f3183#r2077983052